### PR TITLE
Release 106.0 with .NET Standard 2.0 support

### LINKED ISSUE #992: Support .NET Standard 2.0 and .NET Framework 4.5.2 only
This is an epic to combine all related issues

### PR DESCRIPTION
**Pull request by alexeyzimarev**
Release 106.0 with .NET Standard 2.0 support. Fixes #992

### CODE REVIEW

**Comment by alexeyzimarev** [comment]
The check fails since it uses the old master branch build configuration, which is incompatible with the new version of the master branch.

File: RestSharp/Authenticators/HttpBasicAuthenticator.cs [hunk 1]
```diff
@@ -44,9 +44,7 @@ public void Authenticate(IRestClient client, IRestRequest request)
 
             // only add the Authorization parameter if it hasn't been added by a previous Execute
             if (!request.Parameters.Any(p => p.Name.Equals("Authorization", StringComparison.OrdinalIgnoreCase)))
-            {
-                request.AddParameter("Authorization", this.authHeader, ParameterType.HttpHeader);
-            }
+                request.AddParameter("Authorization", authHeader, ParameterType.HttpHeader);
```
Comment (mjwsteenbergen) [inline RestSharp/Authenticators/HttpBasicAuthenticator.cs]: Is it not a better coding convention to always have curly braces after an if?

File: RestSharp/Authenticators/JwtAuthenticator.cs [hunk 2]
```diff
@@ -33,21 +33,17 @@ public class JwtAuthenticator : IAuthenticator
...
 
         public void Authenticate(IRestClient client, IRestRequest request)
         {
             // only add the Authorization parameter if it hasn't been added by a previous Execute
             if (!request.Parameters.Any(p => p.Type.Equals(ParameterType.HttpHeader) &&
                                              p.Name.Equals("Authorization", StringComparison.OrdinalIgnoreCase)))
-            {
-                request.AddParameter("Authorization", this.authHeader, ParameterType.HttpHeader);
-            }
+                request.AddParameter("Authorization", authHeader, ParameterType.HttpHeader);
```
Comment (mjwsteenbergen) [inline RestSharp/Authenticators/JwtAuthenticator.cs]: Is it not a better coding convention to always have curly braces after an if?

File: RestSharp/Authenticators/OAuth/OAuthTools.cs [hunk 3]
```diff
@@ -34,93 +24,85 @@ internal static class OAuthTools
...
         private static readonly object randomLock = new object();
 
-#if !SILVERLIGHT && !WINDOWS_PHONE && !WINDOWS_UWP
         private static readonly RandomNumberGenerator rng = RandomNumberGenerator.Create();
-#endif
+
+        /// <summary>
+        ///     All text parameters are UTF-8 encoded (per section 5.1).
+        /// </summary>
+        /// <seealso cref="http://www.hueniverse.com/hueniverse/2008/10/beginners-gui-1.html" />
```
Comment (mjwsteenbergen) [inline RestSharp/Authenticators/OAuth/OAuthTools.cs]: This link is dead.

File: RestSharp/Authenticators/OAuth/OAuthWorkflow.cs [hunk 4]
```diff
@@ -250,169 +230,130 @@ public virtual OAuthWebQueryInfo BuildClientAuthAccessTokenInfo(string method, W
...
+            };
 
             return info;
         }
 
         private void ValidateTokenRequestState()
         {
-            if (this.RequestTokenUrl.IsNullOrBlank())
-            {
+            if (RequestTokenUrl.IsNullOrBlank())
```
Comment (mjwsteenbergen) [inline RestSharp/Authenticators/OAuth/OAuthWorkflow.cs]: Is it not a better coding convention to always have curly braces after an if? (for this entire class)

File: RestSharp/RestClient.cs [hunk 5]
```diff
@@ -390,10 +294,50 @@ public Uri BuildUri(IRestRequest request)
             return new Uri(assembled);
         }
 
+        /// <summary>
+        ///     Retrieve the handler for the specified MIME content type
+        /// </summary>
+        /// <param name="contentType">MIME content type to retrieve</param>
+        /// <returns>IDeserializer instance</returns>
+        private IDeserializer GetHandler(string contentType)
```
Comment (mjwsteenbergen) [inline RestSharp/RestClient.cs]: Is/should this method be tested?

File: RestSharp/RestRequest.cs [hunk 6]
```diff
@@ -21,98 +21,69 @@
...
         }
 
         /// <summary>
-        /// Sets Resource property
+        ///     Sets Resource property
         /// </summary>
         /// <param name="resource">Resource to use for this request</param>
-        public RestRequest(Uri resource) : this(resource, Method.GET) { }
+        public RestRequest(Uri resource) : this(resource, Method.GET)
+        {
```
Comment (mjwsteenbergen) [inline RestSharp/RestRequest.cs]: Is this better coding convention?

File: RestSharp/RestRequest.cs [hunk 7]
```diff
@@ -368,79 +360,79 @@ public IRestRequest AddObject(object obj, params string[] includedProperties)
...
         public IRestRequest AddParameter(string name, object value, ParameterType type)
         {
-            return this.AddParameter(new Parameter
-                                     {
-                                         Name = name,
-                                         Value = value,
-                                         Type = type
-                                     });
+            return AddParameter(new Parameter
+            {
```
Comment (mjwsteenbergen) [inline RestSharp/RestRequest.cs]: Why was there so much weird formatting? I see that you fixed it all over the place.

File: build.bat [hunk 8]
```diff
@@ -1,119 +1,4 @@
...
-if not "%errorlevel%"=="0" goto failure
-
-:success
-
-exit /B 0
-
-:failure
-
-exit /B -1
+msbuild.exe /t:restore
```
Comment (mjwsteenbergen) [inline build.bat]: This is an improvement 😲

File: RestSharp.IntegrationTests/AuthenticationTests.cs [hunk 9]
```diff
@@ -96,70 +77,22 @@ public void Can_Authenticate_With_OAuth()
             Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
         }
 
-        //[Test]
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply RestSharp.IntegrationTests/AuthenticationTests.cs]: No, I have no idea. This was not in my focus.

[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply RestSharp.IntegrationTests/AuthenticationTests.cs]: I don't like commented out code. If it should be re-written - someone should do it. There is no need for commented out code.

File: RestSharp.IntegrationTests/RestSharp.IntegrationTests.csproj [hunk 10]
```diff
@@ -1,121 +1,33 @@
...
+    <PackageReference Include="NUnit3TestAdapter" Version="3.8.0" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="App.config" />
-    <None Include="packages.config" />
+    <ProjectReference Include="..\RestSharp\RestSharp.csproj" />
   </ItemGroup>
   <ItemGroup>
-    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+    <None Update="Assets\Koala.jpg">
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply RestSharp.IntegrationTests/RestSharp.IntegrationTests.csproj]: Yes, assets are part of the solution.

File: RestSharp.IntegrationTests/oAuth1Tests.cs [hunk 11]
```diff
@@ -383,31 +330,65 @@ public void Can_Query_Vimeo()
...
+            // arrange
+            var client = new RestClient
             {
-                "oauth_consumer_key",
-                "oauth_nonce",
-                "oauth_signature_method",
-                "oauth_timestamp",
-                "oauth_version",
-                "oauth_signature"
+                BaseUrl = new Uri("http://api.linkedin.com/v1"),
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply RestSharp.IntegrationTests/oAuth1Tests.cs]: Don't ask me. I have done nothing there. This is just reformatting.

File: RestSharp.Tests/RestSharp.Tests.csproj [hunk 12]
```diff
@@ -1,245 +1,108 @@
...
-    </ProjectReference>
-    <ProjectReference Include="..\RestSharp\RestSharp.csproj">
-      <Project>{2ececfbf-5f3e-40ee-a963-72336dc7abe2}</Project>
-      <Name>RestSharp</Name>
-    </ProjectReference>
+    <Compile Remove="NuSpecUpdateTask.cs" />
   </ItemGroup>
   <ItemGroup>
-    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+    <None Update="SampleData\4sq.txt">
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply RestSharp.Tests/RestSharp.Tests.csproj]: I don't like to do things implicitly. In addition, I just added them in VS.

File: RestSharp/Authenticators/HttpBasicAuthenticator.cs (hunk 1 above)
Comment (alexeyzimarev) [reply RestSharp/Authenticators/HttpBasicAuthenticator.cs]: Why is it better? It just makes the code longer.

File: RestSharp/Authenticators/OAuth/OAuthTools.cs (hunk 3 above)
Comment (alexeyzimarev) [reply RestSharp/Authenticators/OAuth/OAuthTools.cs]: Sorry, not my fault. It was there and I have done nothing.

File: RestSharp/RestClient.cs (hunk 5 above)
Comment (alexeyzimarev) [reply RestSharp/RestClient.cs]: May be but this is not something I created.

File: RestSharp/RestRequest.cs (hunk 7 above)
Comment (alexeyzimarev) [reply RestSharp/RestRequest.cs]: People used tabs. Tabs are evil.

**Comment by alexeyzimarev** [comment]
My answers in short:
1) Some files were re-formatted and therefore methods appear to be "added". I have not added stuff there. I added some stuff in tests only. Thefefore questions about adding tests aren't really relevant at this point since I have no added new code.
2) Using `var` when explicit type is constructed is a good practice to avoid repetition. Also, many times the type is too long. Many languages do not use explicit types and use inferred types, it does not affect readability.
3) In general, I followed the default Resharper suggestions, which in my opinion contain one of the most comprehensive naming/formatting conventions.
4) I have added ignore for one test. No other tests were changed (except of infrastructural changes), added or removed.

**Comment by alexeyzimarev** [comment]
Just to add to my comments:
My opinion on using `var` is the same as the first answer here: https://stackoverflow.com/questions/41479/use-of-var-keyword-in-c-sharp

Using explicit type in most cases _does not_ improve readability, it just adds noise and makes code harder to read, especially if type is very long, like `IEnumerable<ConcurrentDictionary<string, SomethingVeryImportant>>`.

File: RestSharp/Authenticators/HttpBasicAuthenticator.cs (hunk 1 above)
Comment (mjwsteenbergen) [reply RestSharp/Authenticators/HttpBasicAuthenticator.cs]: It forces that:
```
if(true)
     doThis()
     doThat()
```
will never be interpreted as:
```
if(true)
{
     doThis()
     doThat()
}
```

**Comment by mjwsteenbergen** [comment]
In many cases `var` can increase readability, but in [this case](https://github.com/restsharp/RestSharp/pull/1025/files#diff-38891987c163ee2d8857c8fb0997e988R70) for example, I would not be able to know if `var bytes = s.GetBytes();` returns a List<Bytes>, byte,, byte[], Dictionary<T, byte> or something else.

**Comment by mjwsteenbergen** [comment]
> Some files were re-formatted and therefore methods appear to be "added". I have not added stuff there. I added some stuff in tests only. Thefefore questions about adding tests aren't really relevant at this point since I have no added new code.

Ok, I thought you changed things. I will create issues for the deleted tests and other thing to keep record of it.

**Comment by alexeyzimarev** [comment]
I will check to change R# rules so one-line `if`s will not be changed. May be I need to include a `.DotSettings` file to enforce conventions. We have this in MassTransit.

**Pull request merged**